.github/workflows/staging_deploy.yml

```diff
@@ -34,17 +46,37 @@ jobs:
             sed -i 's|POOL_SIZE=.*|POOL_SIZE=${{ secrets.STAGING_POOL_SIZE }}|' .env
             sed -i 's|PORT=.*|PORT=${{ secrets.STAGING_PORT }}|' .env
             sed -i 's|BEARER_TOKEN=.*|BEARER_TOKEN=${{ secrets.STAGING_BEARER_TOKEN }}|' .env
-            cd /var/www/html
-            sudo ./kill_process_on_port_4000.sh
+            cd /var/www/html &&
+            sudo ./kill_process_on_port_4000.sh || true &&
+            echo '=== Killing any existing beam processes ===' &&
+            (sudo killall beam.smp 2>/dev/null || true) &&
+            sleep 2 &&
             cd /var/www/html/db-service &&
+            echo '=== Debug: Current directory ===' &&
+            pwd &&
+            echo '=== Debug: Checking _build age before clean ===' &&
+            ls -la _build 2>/dev/null || echo 'No _build directory' &&
             sudo git checkout . &&
             if [ -f /var/www/html/db-service/priv/static/swagger.json ]; then
               sudo rm /var/www/html/db-service/priv/static/swagger.json
             fi
-            sudo git pull origin main &&
-            sudo MIX_ENV=prod mix deps.get &&
-            sudo MIX_ENV=prod mix deps.compile &&
-            sudo MIX_ENV=prod mix ecto.migrate &&
-            sudo MIX_ENV=prod mix phx.swagger.generate &&
-            sudo MIX_ENV=prod elixir --erl "-detached" -S mix phx.server
+            sudo git fetch origin &&
+            sudo git checkout $DEPLOY_BRANCH &&
+            sudo git pull origin $DEPLOY_BRANCH &&
+            sudo rm -rf _build deps &&
+            echo '=== Debug: After clean, checking _build ===' &&
+            ls -la _build 2>/dev/null || echo '_build removed successfully' &&
+            echo '=== Debug: Verifying MIX_ENV will be prod ===' &&
+            sudo env MIX_ENV=prod mix --version &&
+            sudo env MIX_ENV=prod mix deps.get &&
+            sudo env MIX_ENV=prod mix deps.compile &&
+            sudo env MIX_ENV=prod mix ecto.migrate || true &&
+            sudo env MIX_ENV=prod mix phx.swagger.generate &&
+            sudo pkill -f 'beam.smp.*mix phx.server' || true &&
```

Comment:
This is redundant and risky. pkill `beam.smp` terminates all Erlang processes on the host, which will not scale if we introduce other BEAM-based services (e.g., another Phoenix app or any other service replying on beam).

A safer alternative is to terminate only the PID bound to port 4000 (as staging works on this port)